src/txmempool.h

```diff
@@ -559,7 +559,7 @@ class CTxMemPool
      * Check that none of this transactions inputs are in the mempool, and thus
      * the tx is not dependent on other mempool transactions to be included in a block.
      */
-    bool HasNoInputsOf(const CTransaction& tx) const;
+    bool HasNoInputsOf(const CTransaction& tx) const EXCLUSIVE_LOCKS_REQUIRED(cs);
```

Comment:
Would it be clearer to just add `LOCK(cs);` to the top of [the function definition](https://github.com/jamesob/bitcoin/blob/81bd349c9c8dd247d32ece1adcc6db1254ef8971/src/txmempool.cpp#L885-L886) instead of this annotation since there's a recursive lock acquisition in `CTxMempool::exists` anyway?

Reply:
Imo it is clearer to let the caller figure out how long the lock is required to be held. Imagine this is called in a loop, the you, most likely, wouldn't want to release the lock in between each iteration? Also, it happens that all callers already have the lock taken, since this annotation can be added without any other changes.

Reply:
Yep, makes sense.